Developing the nerwann module to calculate thermomagnetic coefficients

Emad-Rezaei opened the pull request:
Developing the nerwann module to calculate thermomagnetic coefficients

Emad-Rezaei committed:
Update Makefile.2

Emad-Rezaei committed:
Update Makefile.header

Emad-Rezaei committed:
Update readwrite.F90

Emad-Rezaei committed:
Update postw90_common.F90

Emad-Rezaei committed:
Update postw90.F90

Emad-Rezaei committed:
Update postw90_readwrite.F90

Emad-Rezaei committed:
Update postw90_types.F90

Emad-Rezaei committed:
Update wan_ham.F90

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Update CHANGELOG.md

Emad-Rezaei committed:
Update user_guide.tex

Emad-Rezaei committed:
Create nerwann.tex

Emad-Rezaei committed:
Update postw90params.tex

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Update nerwann.F90

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Update nerwann.tex

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Update postw90params.tex

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Update ZnSe.win

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Update nerwann.tex

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Add files via upload

Emad-Rezaei committed:
Merge pull request #1 from Emad-Rezaei/Emad-Rezaei-patch-1

Developing the nerwann module to calculate thermomagnetic coefficients

Emad-Rezaei committed:
Update nerwann.F90

Emad-Rezaei committed:
Update constants.F90

Adding "dkpt" for the finite difference method

Emad-Rezaei committed:
Update Makefile.2

Emad-Rezaei committed:
Update wannier90_readwrite.F90

stepan-tsirkin commented on src/postw90/nerwann.F90:
```diff
@@ -0,0 +1,1203 @@
+        end if
+
+!  write(*,6)'loop,rdum,occ,vel=',loop_f,rdum,r_num_elec_per_state,vel(BandIdx,1:3) !Added by Dr.Keivan main write
+!Total TDF 1st+2nd along Bz
```
The following lines will become more readable if you pre-define some constants and then use them, like 

```
rdum1 = rdum*(pw90_nerwann%relax_time**(-1))*r_num_elec_per_state*& 
                        physics%elem_charge_SI**3/physics%hbar_SI**2*1.e-5_dp
rdum2 = rdum* r_num_elec_per_state*physics%elem_charge_SI**4/ & 
                                physics%hbar_SI**3*1.e-40_dp
```

Then the next line reads just 
```
TDFtot_kz(XX, loop_f, 1) = TDFtot_kz(XX, loop_f,1)+rdum1*vel_k(BandIdx,1)*vel_k(BandIdx,1)+&
                  rdum2*vel_k(BandIdx,1)*omg_bnd1(BandIdx)
```

Further,  combining the 3 arrays `omg_bnd1` ,`omg_bnd2`, `omg_bnd3` into one 2d-adday, the next 9 lines can be rewritten as a double-nested loop. 

Same aapplies till line 1192

Emad-Rezaei committed:
Add files via upload

updating postw90_common.F90, postw90_readwrite.F90, postw90_types.F90, and nerwann.F90

Emad-Rezaei committed:
Add files via upload

updating readwrite.F90

Emad-Rezaei committed:
Add files via upload

adding InSb as a test

Emad-Rezaei committed:
Delete test-suite/tests/testpostw90_nerwann directory

Emad-Rezaei committed:
Add files via upload

adding test testpostw90_nerwann

Emad-Rezaei committed:
Delete test-suite/checkpoints/znse_postw90 directory

Emad-Rezaei committed:
Add files via upload

uploading a check test for nerwann

Emad-Rezaei committed:
Add files via upload

modifying userconfig and jobconfig files

Emad-Rezaei committed:
Add files via upload

parser for Nernst

Emad-Rezaei committed:
Add files via upload

wan_ham.F90 updated.

Emad-Rezaei committed:
Update nerwann.tex

Emad-Rezaei committed:
Update wannier90.bib

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Update postw90params.tex

Emad-Rezaei committed:
Update tutorial.tex

Emad-Rezaei committed:
Delete ZnSe.nscf

Emad-Rezaei committed:
Delete ZnSe.pw2wan

Emad-Rezaei committed:
Delete ZnSe.scf

Emad-Rezaei committed:
Delete ZnSe.win

Emad-Rezaei committed:
Update README

Thermomagnetic effect for InSb

Emad-Rezaei committed:
Add files via upload

Nerwann example

Emad-Rezaei committed:
Add files via upload

Makefile.2 and Makefile.header
